File: js/smallerThanNeighbours.js
function returnPeakElement(nums){
    var max =0;
    for(let i =1; i < nums.length-1; i++){
        var element = nums[i];
        var first = nums[i-1];
        var second = nums[i+1];

        if(element > first && element > second){
            if(max < element) max = element;
        }
    }
    return max;
}

console.log(returnPeakElement([1,1,5,4,9,2]));

File: java/SmallerThanNeighbours.java


public class SmallerThanNeighbours {
    public static void main(String[] args) {
        System.out.println(returnPeakElement(new int[]{1,5,9,3}));
    }

    public static int returnPeakElement(int [] nums){
        int max =0;
        for(int i = 1; i < nums.length-1; i++){
            int first = nums[i-1];
            int mid = nums[i];
            int second = nums[i+1];
            if(mid > first && mid > second){
                if(mid > max){
                    max = mid;
                }
            }
    
        }    
        return max;
    }
}
